src/app/geonode/qgsgeonodenewconnection.cpp

```diff
@@ -0,0 +1,293 @@
+      break;
+    case 2: // QGIS
+      dpiMode = 1;
+      break;
+    case 3: // UMN
+      dpiMode = 2;
+      break;
+    case 4: // GeoServer
+      dpiMode = 4;
+      break;
+  }
+
+  settings.setValue( key + "/wms/dpiMode", dpiMode );
+  settings.setValue( key + "/wms/referer", txtReferer->text() );
+
+  QString version = QStringLiteral( "auto" );
+  switch ( cmbVersion->currentIndex() )
+  {
+    case 0:
+      version = QStringLiteral( "auto" );
+      break;
+    case 1:
+      version = QStringLiteral( "1.0.0" );
+      break;
+    case 2:
+      version = QStringLiteral( "1.1.0" );
+      break;
+    case 3:
+      version = QStringLiteral( "2.0.0" );
+      break;
+  }
+
+  settings.setValue( key + "/wfs/version", version );
+  settings.setValue( key + "/wfs/maxnumfeatures", txtMaxNumFeatures->text() );
+
+  settings.setValue( credentialsKey + "/username", txtUserName->text() );
+  settings.setValue( credentialsKey + "/password", txtPassword->text() );
+
+  settings.setValue( credentialsKey + "/authcfg", mAuthConfigSelect->configId() );
+
+  settings.setValue( mBaseKey + "/selected", txtName->text() );
+
+  QDialog::accept();
+}
+
+void QgsGeoNodeNewConnection::okButtonBehavior( const QString &text )
+{
+  Q_UNUSED( text );
+  buttonBox->button( QDialogButtonBox::Ok )->setDisabled( txtName->text().isEmpty() || txtUrl->text().isEmpty() );
+  buttonBox->button( QDialogButtonBox::Ok )->setEnabled( !txtName->text().isEmpty() && !txtUrl->text().isEmpty() );
+}
+
+void QgsGeoNodeNewConnection::testConnection()
+{
+  QApplication::setOverrideCursor( Qt::BusyCursor );
+  QString endpoint( "/api/layers/" );
+  QNetworkReply *layersReply = request( endpoint );
+  endpoint = "/api/maps";
+  QNetworkReply *mapsReply = request( endpoint );
+  QApplication::restoreOverrideCursor();
+
+  if ( layersReply->error() == QNetworkReply::NoError && mapsReply->error() == QNetworkReply::NoError )
+  {
+    QMessageBox::information( this,
+                              tr( "Test connection" ),
+                              tr( "\nConnection to %1 was successful, \n\n%1 is a valid geonode instance.\n\n" ).arg( txtUrl->text() ) );
+  }
+  else
+  {
+    QMessageBox::information( this,
+                              tr( "Test connection" ),
+                              tr( "\nConnection failed, \n\nplease check whether %1 is a valid geonode instance.\n\n" ).arg( txtUrl->text() ) );
+  }
+}
+
+QNetworkReply *QgsGeoNodeNewConnection::request( QString &endPoint )
+{
+  QString url = txtUrl->text() + endPoint;
+  if ( !url.contains( QLatin1String( "://" ) ) )
+  {
+    url.prepend( "http://" );
+  }
+
+  QUrl layerUrl( url );
+  layerUrl.setScheme( "http" );
+  QgsNetworkAccessManager *networkManager = QgsNetworkAccessManager::instance();
+
+  QNetworkRequest request( layerUrl );
+  request.setHeader( QNetworkRequest::ContentTypeHeader, "application/json" );
+
+  QNetworkReply *reply = networkManager->get( request );
+  while ( !reply->isFinished() )
```

Comment:
Can you make this asynchroneous by connecting to the `QNetworkReply::finished` signal?
I have seen too much troubles with `processEvents()` (crashes which just happen randomly and are hard to track down).